Common/Classing/ClassingPlayer.cs

```diff
@@ -0,0 +1,50 @@
+﻿using PathOfTerraria.Common.Looting.VirtualBagUI;
+using PathOfTerraria.Core.UI;
+using Terraria.Localization;
+using Terraria.ModLoader.IO;
+using Terraria.UI;
+
+namespace PathOfTerraria.Common.Classing;
+
+internal class ClassingPlayer : ModPlayer
+{
+	public StarterClasses Class = StarterClasses.None;
+
+	/// <summary>
+	/// Gets the class noun, i.e. Warrior or Marksman. Returns "Stranger" if the player has no selected class.
+	/// </summary>
+	public static string GetClassNoun(Player player, int index = 0)
+	{
+		ClassingPlayer plr = player.GetModPlayer<ClassingPlayer>();
+		string noun;
+
+		if (plr.Class == StarterClasses.None)
+		{
+			noun = Language.GetTextValue("Mods.PathOfTerraria.UI.ClassPages.DefaultNouns");
+		}
+		else
+		{
+			noun = plr.Class.GetNouns();
+		}
+
+		return noun.Split(' ')[index];
+	}
```

Comment:
This has two issues:
- The index should be wrapped (`idx % arr.Length`) to prevent localization-caused IOOB errors. Worth logging when that happens.
- Using spaces as a separator is not language-neutral, some languages are sure to require multiple words for what is a single one in English. A semicolon should be better.